readme.md

```diff
@@ -343,15 +343,60 @@ for ($i = 0; $i < 10; $i++) {
 print_r($values); // [1, 4, null, 9, 5, null, null, 4, 6, null]
 
 // optional() accepts a weight argument to specify the probability of receiving the default value.
-// 0 will always return the default value; 1 will always return the provider. Default weight is 0.5 (50% chance).
-$faker->optional($weight = 0.1)->randomDigit; // 90% chance of NULL
-$faker->optional($weight = 0.9)->randomDigit; // 10% chance of NULL
+// --------------------------------------------------------
+// 2 value sets are possible
+// --------------------------------------------------------
+// First set is between 0 and 1 (Fractional representation)
+// --------------------------------------------------------
+// 0 will always return the default value
+// Anything else BELOW 1 will be converted to a percentage
+// 
+// --------------------------------------------------------
+// Second set is between 1 and 100 (Percentage representation)
```

Comment:
Perhaps we should add a note that this behaviour will be removed in `2.0`?

See https://github.com/fzaninotto/Faker/blob/3f74f49dd7c53af959b3708a64d9232db271f985/src/Faker/Provider/Base.php#L547-L551.

Reply:
I disagree. This kind of mention should only appear in the documentation in an Upgrade guide for 2.0. No need to add mental overhead to readers about something that doesn't concern them.